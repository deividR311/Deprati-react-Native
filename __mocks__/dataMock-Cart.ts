export const DataMock_Cart = {
  carts: [
    {
      appliedOrderPromotions: [],
      appliedProductPromotions: [],
      appliedVouchers: [],
      calculated: true,
      code: '4809002',
      deliveryAddress: {
        cellphoneNumber: '34949494',
        cellphonePreffix: '09',
        city: 'COT-287',
        country: {
          isocode: 'EC',
          name: 'Ecuador'
        },
        defaultAddress: true,
        email: 'alexis.enriquezca@gmail.com',
        firstName: 'Alejandra ',
        formattedAddress: 'Calle 6, Casa 23, Cotopaxi, Alaquez',
        id: '8819555172375',
        idNumber: '1721480299',
        idType: 'CEDULA',
        line1: 'Calle 6',
        line2: 'Casa 23',
        otherInfo: 'Laura',
        phone: '',
        region: {
          countryIso: 'EC',
          isocode: 'EC-COT',
          isocodeShort: '193',
          name: 'Cotopaxi'
        },
        shippingAddress: true,
        town: 'Alaquez',
        visibleInAddressBook: true
      },
      deliveryItemsQuantity: 1,
      deliveryOrderGroups: [
        {
          entries: [
            {
              basePrice: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              basePriceWithTaxes: {
                currencyIso: 'USD',
                formattedValue: '$2,60',
                priceType: 'BUY',
                value: 2.5984
              },
              basePriceWithoutTaxes: {
                currencyIso: 'USD',
                formattedValue: '$23,20',
                priceType: 'BUY',
                value: 23.2
              },
              deliveryMode: {
                code: 'standard'
              },
              deliveryTimeRange: '5 a 7 días laborables',
              entryNumber: 0,
              giftPacking: false,
              pickup: false,
              product: {
                acceptGiftPacking: true,
                availableForPickup: false,
                baseOptions: [
                  {
                    selected: {
                      code: '226010190802030008',
                      priceData: {
                        currencyIso: 'USD',
                        formattedValue: '$2,32',
                        priceType: 'BUY',
                        value: 2.32
                      },
                      stock: {
                        stockLevel: 11,
                        stockLevelStatus: 'inStock'
                      },
                      url: '/camiseta-estampada/p/226010190802030008',
                      variantOptionQualifiers: []
                    },
                    variantType: 'GenericVariantProduct'
                  }
                ],
                baseProduct: '000226010190802030',
                categories: [
                  {
                    code: 'T8023',
                    name: '10',
                    parentCategoryName: 'Talla',
                    url: '/10/c/T8023'
                  },
                  {
                    code: 'C0110601',
                    name: 'Blanco',
                    parentCategoryName: 'Color',
                    url: '/blanco/c/C0110601'
                  },
                  {
                    code: '0701',
                    name: 'MUJERES',
                    url: '/mujeres/c/0701'
                  }
                ],
                code: '226010190802030008',
                deliveryModes: [
                  {
                    code: 'standard'
                  },
                  {
                    code: 'pickup'
                  }
                ],
                ean: '15282579',
                images: [
                  {
                    format: 'zoom',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h67/h07/8890805157918/15282579-0_product_515Wx772H'
                  },
                  {
                    format: 'product',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h26/h0b/8890805288990/15282579-0_product_300Wx450H'
                  },
                  {
                    format: 'thumbnail',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h38/h0e/8890805354526/15282579-0_product_96Wx144H'
                  },
                  {
                    format: 'cartIcon',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h49/h11/8890805420062/15282579-0_product_65Wx97H'
                  }
                ],
                isGiftProduct: false,
                isPromotionSpecialPrice: false,
                ivaCondition: '12%',
                name: 'CAMISETA ESTAMPADA',
                purchasable: true,
                stock: {
                  stockLevel: 11,
                  stockLevelStatus: 'inStock'
                },
                url: '/camiseta-estampada/p/226010190802030008'
              },
              quantity: 1,
              subtotal: {
                currencyIso: 'USD',
                formattedValue: '$23,20',
                priceType: 'BUY',
                value: 23.2
              },
              totalDiscount: {
                currencyIso: 'USD',
                formattedValue: '$20,88',
                priceType: 'BUY',
                value: 20.88
              },
              totalPrice: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              totalPriceWithoutTaxes: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              updateable: true,
              volume: 0,
              weight: 0
            }
          ],
          totalPriceWithTax: {
            currencyIso: 'USD',
            formattedValue: '$2,60',
            priceType: 'BUY',
            value: 2.5999999999999996
          }
        }
      ],
      entries: [
        {
          basePrice: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          basePriceWithTaxes: {
            currencyIso: 'USD',
            formattedValue: '$2,60',
            priceType: 'BUY',
            value: 2.5984
          },
          basePriceWithoutTaxes: {
            currencyIso: 'USD',
            formattedValue: '$23,20',
            priceType: 'BUY',
            value: 23.2
          },
          deliveryMode: {
            code: 'standard'
          },
          deliveryTimeRange: '5 a 7 días laborables',
          entryNumber: 0,
          giftPacking: false,
          pickup: false,
          product: {
            acceptGiftPacking: true,
            availableForPickup: false,
            baseOptions: [
              {
                selected: {
                  code: '226010190802030008',
                  priceData: {
                    currencyIso: 'USD',
                    formattedValue: '$2,32',
                    priceType: 'BUY',
                    value: 2.32
                  },
                  stock: {
                    stockLevel: 11,
                    stockLevelStatus: 'inStock'
                  },
                  url: '/camiseta-estampada/p/226010190802030008',
                  variantOptionQualifiers: []
                },
                variantType: 'GenericVariantProduct'
              }
            ],
            baseProduct: '000226010190802030',
            categories: [
              {
                code: 'T8023',
                name: '10',
                parentCategoryName: 'Talla',
                url: '/10/c/T8023'
              },
              {
                code: 'C0110601',
                name: 'Blanco',
                parentCategoryName: 'Color',
                url: '/blanco/c/C0110601'
              },
              {
                code: '0701',
                name: 'MUJERES',
                url: '/mujeres/c/0701'
              }
            ],
            code: '226010190802030008',
            deliveryModes: [
              {
                code: 'standard'
              },
              {
                code: 'pickup'
              }
            ],
            ean: '15282579',
            images: [
              {
                format: 'zoom',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h67/h07/8890805157918/15282579-0_product_515Wx772H'
              },
              {
                format: 'product',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h26/h0b/8890805288990/15282579-0_product_300Wx450H'
              },
              {
                format: 'thumbnail',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h38/h0e/8890805354526/15282579-0_product_96Wx144H'
              },
              {
                format: 'cartIcon',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h49/h11/8890805420062/15282579-0_product_65Wx97H'
              }
            ],
            isGiftProduct: false,
            isPromotionSpecialPrice: false,
            ivaCondition: '12%',
            name: 'CAMISETA ESTAMPADA',
            purchasable: true,
            stock: {
              stockLevel: 11,
              stockLevelStatus: 'inStock'
            },
            url: '/camiseta-estampada/p/226010190802030008'
          },
          quantity: 1,
          subtotal: {
            currencyIso: 'USD',
            formattedValue: '$23,20',
            priceType: 'BUY',
            value: 23.2
          },
          totalDiscount: {
            currencyIso: 'USD',
            formattedValue: '$20,88',
            priceType: 'BUY',
            value: 20.88
          },
          totalPrice: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          totalPriceWithoutTaxes: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          updateable: true,
          volume: 0,
          weight: 0
        }
      ],
      guid: '15901dbd-9966-4239-9a4d-fecc4f747589',
      net: true,
      orderDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      pickupItemsQuantity: 0,
      pickupOrderGroups: [],
      productDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      sellWithTaxes: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      sellWithoutTaxWithoutInterest: {
        currencyIso: 'USD',
        formattedValue: '$2,31',
        priceType: 'BUY',
        value: 2.31
      },
      sellWithoutTaxes: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      site: 'deprati',
      store: 'deprati',
      subTotal: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      totalDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$20,88',
        priceType: 'BUY',
        value: 20.88
      },
      totalItems: 1,
      totalPrice: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      totalPriceWithTax: {
        currencyIso: 'USD',
        formattedValue: '$2,60',
        priceType: 'BUY',
        value: 2.6
      },
      totalTax: {
        currencyIso: 'USD',
        formattedValue: '$0,28',
        priceType: 'BUY',
        value: 0.28
      },
      totalVolume: 0,
      totalWeight: 0.25,
      user: {
        name: 'Alexis pruebas ENRIQUEZ',
        uid: 'alexis.enriquezca@gmail.com'
      },
      potentialOrderPromotions: [
        {
          consumedEntries: [],
          promotion: {
            code: '10012022_EGCE_PRUEBAAPP1_A',
            description: 'Version 2 Hybris - PRUEBA APP',
            endDate: '2024-01-01T04:59:59+0000',
            promotionType: 'Promoción basada en regla'
          }
        },
        {
          consumedEntries: [],
          description:
            'Completa $150 y  obten envío gratis. Aplica restricciones.',
          promotion: {
            code: '10012022_EGCE_PRUEBAAPP1_C',
            description: 'Version 2 Hybris',
            endDate: '2024-01-01T04:59:59+0000',
            promotionType: 'Promoción basada en regla'
          }
        }
      ],
      potentialProductPromotions: [],
      totalUnitCount: 1
    }
  ]
};
