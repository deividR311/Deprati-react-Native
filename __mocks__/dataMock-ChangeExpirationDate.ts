export const DataMock_ChangeExpirationDate = {
  TEXT_NEWDATE: 'Selecciona el día',
  infoCreditBalance: {
    clientName: 'PATRICIA L. DE UGALDE',
    cardNumber: '1xxxxx6',
    affiliateDate: '2000',
    totalAmount: '5.500,00',
    amountSpent: '724,35',
    aditional: '00',
    availableAmount: '4.775,65',
    pastDueDays: 0,
    nextPayDay: 'Abril 14 de 2023',
    nextAmountToPay: '105,08',
    numeroTarjetaAdicionalDisplay: '14XXX86 00',
    clientFirstName: 'Prueba'
  },
  dataCurrentDate: {
    cuttingDay: 14,
    dueDate: '20230514',
    day: '14'
  },
  dataChangeDate: null,
  selectedDay: '2',
  isAcceptTerms: false,
  showWarning: false,
  isLoadingChangeDate: false,
  isLoadingCurrentDate: false,
  showSucces: false,
  showErrorSucces: false,
  expirationDateContent: {
    legend:
      'No puedes cambiar tu fecha máxima de pago si tienes una deuda pendiente y <b>este cambio se hará efectivo a partir del próximo mes.</b>',
    popup_title:
      '¿Estás seguro de cambiar la fecha máxima de pago de tu crédito De Prati?',
    popup_paragraphs: [
      'Ten en cuenta:',
      '1. Esta acción no se puede deshacer y el cambio se hará efectivo a partir de tu próximo mes.',
      '2. Si haces el cambio de fecha <b>deberás pagar la cuota del mes en curso</b> y este cambio se hará efectivo a partir del próximo mes.'
    ],
    popup_accept_action: 'SI, ESTOY DE ACUERDO',
    popup_cancel_action: 'NO, CANCELAR'
  },
  selectRef: {
    current: {
      reset: jest.fn(),
      openDropdown: jest.fn(),
      closeDropdown: jest.fn()
    }
  },
  setSelectedDay: jest.fn(),
  setIsAcceptTerms: jest.fn(),
  handleChangeDate: jest.fn(),
  setShowWarning: jest.fn(),
  handleCloseSuccess: jest.fn(),
  handleCloseError: jest.fn(),
  handleCloseWarning: jest.fn()
};
