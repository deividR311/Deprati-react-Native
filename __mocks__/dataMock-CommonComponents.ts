export const DataMock_CustomProps = {
  url: 'url',
  style: {},
  content: 'qwerty',
  uid: 'uid',
  typeCode: 'qwerty',
  modifiedTime: 'qwerty',
  name: 'qwerty',
  container: true,
  rotatingImagesComponentsList: 'DepratiBrandsRotatingImages',
  childrenComponentsList: ['DepratiBrandsRotatingImages'],
  childrenComponents: {
    DepratiBrandsRotatingImages: {
      uid: 'DepratiBrandsRotatingImages',
      typeCode: 'RotatingImagesComponent',
      modifiedTime: '2023-04-28T12:59:58+0000',
      name: 'DePrati Dior Mobile Carousel',
      container: 'false',
      resolutionEnum: 'MOBILE',
      effect: 'FADE',
      banners: 'DiorBannerComponent2 BannerCarouselDiorVideo_1',
      timeout: '6',
      childrenComponentsList: [
        'DiorBannerComponent2',
        'BannerCarouselDiorVideo_1'
      ]
    }
  }
};

export const DataMock_BillingAddressComponent = {
  style: {},
  onSelected: () => {},
  emptyStyle: {},
  showTitle: true,
  enableHandleEnableButton: true,
  onIsEnable: () => {},
  paymentAddress: {
    defaultAddress: false,
    email: 'string',
    firstName: 'string',
    formattedAddress: 'string',
    id: 'string',
    line1: 'string',
    line2: 'string',
    phone: 'string',
    phonePreffix: 'string',
    shippingAddress: false,
    visibleInAddressBook: false
  },
  paymentCredit: {
    showAlerAddressPayment: false,
    setShowAlerAddressPayment: () => {}
  },
  styleSecundaryButton: false
};

export const DataMock_useBillingAddress = {
  addressSelected: {
    defaultAddress: false,
    firstName: 'firstName',
    id: 'qwertystring',
    line1: 'qwerty',
    line2: 'qwerty',
    phone: 'qwerty',
    phonePreffix: 'qwerty'
  },
  isLoadingByDeleting: false,
  addressPaymentList: {
    default: null,
    others: [],
    existsAddress: false,
    loading: false
  },
  isLoading: false,
  buttonChangeData: false,
  handleChangeData: jest.fn(),
  changeAddress: {
    handleSelected: jest.fn(),
    addressSelected: undefined
  },
  handleDelete: jest.fn(),
  handleAddEdit: jest.fn(),
  handleBillingAddressList: () => {
    return [
      {
        defaultAddress: false,
        email: 'string',
        firstName: 'string',
        formattedAddress: 'string',
        id: 'string',
        line1: 'string',
        line2: 'string',
        phone: 'string',
        phonePreffix: 'string',
        shippingAddress: false,
        visibleInAddressBook: false
      }
    ];
  }
};

export const DataMock_IncreaseCreditLimit = {
  rectangularSize: 232,
  value: 1000,
  maxValue: 2000,
  onChangeCreditLimit: jest.fn()
};

export const DataMock_Charts = {
  constentChart: {
    availableAmount: 100,
    amountSpent: 20,
    totalAmount: 200
  },
  ownerFullname: 'Custom customer customization',
  invertValues: true,
  showDisplayCard: true
};
