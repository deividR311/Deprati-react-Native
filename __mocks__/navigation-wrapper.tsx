import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { COLORS } from '../src/application/common';
import { ProvideWrapper } from './provider-wrapper';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

const Stack = createNativeStackNavigator();

export const NavigationWrapper = ({ children, screen, initialParams }: any) => (
  <ProvideWrapper>
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="default"
        screenOptions={{
          headerShown: true,
          headerShadowVisible: false,
          headerTitleAlign: 'left',
          contentStyle: { backgroundColor: COLORS.WHITE }
        }}>
        {screen && (
          <Stack.Screen
            name="default"
            initialParams={initialParams ?? {}}
            component={screen}
          />
        )}
        {children && <Stack.Screen name="default" component={() => children} />}
      </Stack.Navigator>
    </NavigationContainer>
  </ProvideWrapper>
);
