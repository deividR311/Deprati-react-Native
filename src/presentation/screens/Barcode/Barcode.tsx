import QRCodeScanner from 'react-native-qrcode-scanner';
import { BarCodeReadEvent, RNCamera } from 'react-native-camera';

import React, { useEffect } from 'react';
import Icon from 'react-native-vector-icons/MaterialIcons';
import {
  Image,
  SafeAreaView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View
} from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { NAV } from '../../../application/common/namesNav';
import { EcommerceNavigationRoute } from '../../navigation/ecommerce';
import { COLORS, MARING_HORIZONTAL } from '../../../application/common';
import { FontStyles } from '../../../application/common/fonts';

const Barcode = () => {
  const navigation = useNavigation();

  const onEanDetected = ({ data: code, type }: BarCodeReadEvent) => {
    //console.log('>>> EAN detected: ', type, code)
    navigation.replace(NAV.ECOMMERCE, {
      screen: EcommerceNavigationRoute.Searchproduct,
      params: {
        EAN: code
      }
    });
  };
  useEffect(() => {
    if (__DEV__) {
      onEanDetected({ data: '00000014615201' });
    }
  }, []);

  const topContent = () => {
    return (
      <SafeAreaView style={styles.container_header}>
        <TouchableOpacity
          style={styles.container_header_back}
          onPress={() => navigation.goBack()}>
          <Icon name="arrow-back" size={24} color={COLORS.WHITE} />
        </TouchableOpacity>
        <Text style={styles.container_header_text}>Buscar por producto</Text>
        <TouchableOpacity
          style={styles.close}
          onPress={() => navigation.goBack()}>
          <Icon name="close" size={20} color={COLORS.DARK} />
        </TouchableOpacity>
      </SafeAreaView>
    );
  };

  return (
    <QRCodeScanner
      onRead={onEanDetected}
      reactivate={true}
      reactivateTimeout={5000}
      showMarker={true}
      fadeIn={false}
      vibrate={true}
      customMarker={
        <>
          {topContent()}
          <Image
            style={styles.marker}
            source={require('../../../../assets/images/barcode-custom-frame.png')}
          />
        </>
      }
      flashMode={RNCamera.Constants.FlashMode.auto}
      cameraStyle={{
        height: '100%'
      }}
    />
  );
};

export default Barcode;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'flex-start',
    justifyContent: 'flex-start'
  },
  container_header: {
    position: 'absolute',
    top: 0,
    flexDirection: 'row',
    marginVertical: 8,
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
    elevation: 2,
    zIndex: 2000
    // backgroundColor: 'transparent',
  },
  container_header_back: {
    marginLeft: MARING_HORIZONTAL
  },
  container_header_text: {
    ...FontStyles.H3_Headline,
    color: COLORS.WHITE,
    textAlign: 'left',
    paddingHorizontal: MARING_HORIZONTAL,
    paddingVertical: MARING_HORIZONTAL,
    flexGrow: 2
  },
  barcodeText: {
    fontSize: 20,
    color: 'white',
    fontWeight: 'bold'
  },
  close: {
    width: 40,
    height: 40,
    borderTopLeftRadius: 8,
    borderBottomLeftRadius: 8,
    backgroundColor: COLORS.DEPRATYGRAY,
    elevation: 2,
    zIndex: 2,
    alignItems: 'center',
    justifyContent: 'center'
  },
  marker: {
    width: 300,
    height: 300
  }
});
