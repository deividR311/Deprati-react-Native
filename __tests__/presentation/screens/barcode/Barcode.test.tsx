import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationWrapper } from '../../../../__mocks__/navigation-wrapper';
import Barcode from '../../../../src/presentation/screens/Barcode/Barcode';

let screenTest: any;

describe('Barcode', () => {
  beforeEach(() => {
    screenTest = render(<NavigationWrapper screen={Barcode} />);
  });
  it('renders correctly in Snapshot', () => {
    expect(screenTest.toJSON()).toMatchSnapshot();
  });
});
