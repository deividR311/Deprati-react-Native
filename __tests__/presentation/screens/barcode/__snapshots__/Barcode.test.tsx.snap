// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Barcode renders correctly in Snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        hideShadow={true}
        largeTitleHideShadow={false}
        title="default"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
          ]
        }
      >
        <QRCodeScanner
          cameraStyle={
            Object {
              "height": "100%",
            }
          }
          customMarker={
            <React.Fragment>
              <ForwardRef(SafeAreaView)
                style={
                  Object {
                    "alignItems": "center",
                    "elevation": 2,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginVertical": 8,
                    "position": "absolute",
                    "top": 0,
                    "width": "100%",
                    "zIndex": 2000,
                  }
                }
              >
                <TouchableOpacity
                  onPress={[Function]}
                  style={
                    Object {
                      "marginLeft": 16,
                    }
                  }
                >
                  <Icon
                    allowFontScaling={false}
                    color="#ffffff"
                    name="arrow-back"
                    size={24}
                  />
                </TouchableOpacity>
                <Text
                  style={
                    Object {
                      "color": "#ffffff",
                      "flexGrow": 2,
                      "fontFamily": "Roboto-Medium",
                      "fontSize": 20,
                      "fontWeight": "500",
                      "letterSpacing": 0.15,
                      "lineHeight": 24,
                      "paddingHorizontal": 16,
                      "paddingVertical": 16,
                      "textAlign": "left",
                    }
                  }
                >
                  Buscar por producto
                </Text>
                <TouchableOpacity
                  onPress={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#E9EBED",
                      "borderBottomLeftRadius": 8,
                      "borderTopLeftRadius": 8,
                      "elevation": 2,
                      "height": 40,
                      "justifyContent": "center",
                      "width": 40,
                      "zIndex": 2,
                    }
                  }
                >
                  <Icon
                    allowFontScaling={false}
                    color="#000000"
                    name="close"
                    size={20}
                  />
                </TouchableOpacity>
              </ForwardRef(SafeAreaView)>
              <Image
                source={
                  Object {
                    "testUri": "../../../assets/images/barcode-custom-frame.png",
                  }
                }
                style={
                  Object {
                    "height": 300,
                    "width": 300,
                  }
                }
              />
            </React.Fragment>
          }
          fadeIn={false}
          onRead={[Function]}
          reactivate={true}
          reactivateTimeout={5000}
          showMarker={true}
          vibrate={true}
        />
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
